File: Source/TurnTracker.Client/src/app/user/models/EditableActivity.ts
import { User } from './User';
import { Unit } from './Unit';
import { NotificationSetting } from './NotificationSetting';

export class EditableActivity {
    public id: number;
    public name: string;
    public description: string;
    public isDisabled: boolean;
    public periodUnit?: Unit;
    public periodCount?: number;
    public takeTurns: boolean;
    public participants: User[];
    public defaultNotificationSettings: NotificationSetting[];
}

File: Source/TurnTracker.Domain/Models/EditableActivity.cs
using System.Collections.Generic;
using TurnTracker.Data;

namespace TurnTracker.Domain.Models
{
    public class EditableActivity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsDisabled { get; set; }
        public Unit? PeriodUnit { get; set; }
        public uint? PeriodCount { get; set; }
        public bool TakeTurns { get; set; }
        public List<UserInfo> Participants { get; set; }
        public List<NotificationInfo> DefaultNotificationSettings { get; set; }
    }
}